packages/@aws-cdk/aws-ec2/lib/launch-template.ts

```diff
@@ -0,0 +1,688 @@
+   * isn't available with all instance types. Additional usage charges apply when using an EBS-optimized instance.
+   *
+   * @default - EBS optimization is not specified in the launch template.
+   */
+  readonly ebsOptimized?: boolean;
+
+  /**
+   * If this parameter is set to true, the instance is enabled for AWS Nitro Enclaves; otherwise, it is not enabled for AWS Nitro Enclaves.
+   *
+   * @default - Enablement of Nitro enclaves is not specified in the launch template; defaulting to false.
+   */
+  readonly nitroEnclaveEnabled?: boolean;
+
+  /**
+   * If you set this parameter to true, the instance is enabled for hibernation.
+   *
+   * @default - Hibernation configuration is not specified in the launch template; defaulting to false.
+   */
+  readonly hibernationConfigured?: boolean;
+
+  /**
+   * Indicates whether an instance stops or terminates when you initiate shutdown from the instance (using the operating system command for system shutdown).
+   *
+   * @see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/terminating-instances.html#Using_ChangingInstanceInitiatedShutdownBehavior
+   *
+   * @default - Shutdown behavior is not specified in the launch template; defaults to STOP.
+   */
+  readonly instanceInitiatedShutdownBehavior?: InstanceInitiatedShutdownBehavior;
+
+  /**
+   * If this property is defined, then the Launch Template's InstanceMarketOptions will be
+   * set to use Spot instances, and the options for the Spot instances will be as defined.
+   *
+   * @default - Instance launched with this template will not be spot instances.
+   */
+  readonly spotOptions?: LaunchTemplateSpotOptions;
+
+  /**
+   * Name of SSH keypair to grant access to instance
+   *
+   * @default - No SSH access will be possible.
+   */
+  readonly keyName?: string;
+
+  /**
+   * If set to true, then detailed monitoring will be enabled on instances created with this
+   * launch template.
+   *
+   * @see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-cloudwatch-new.html
+   *
+   * @default False - Detailed monitoring is disabled.
+   */
+  readonly detailedMonitoring?: boolean;
+
+  /**
+   * Security group to assign to instances created with the launch template.
+   *
+   * @default No security group is assigned.
+   */
+  readonly securityGroup?: ISecurityGroup;
+
+  /**
+   * The tags to apply to the resources during launch. The specified tags are applied to all resources of the given
+   * types launch.
+   * Note that not every resource supports tagging during creation. See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-tagspecification.html#cfn-ec2-launchtemplate-tagspecification-resourcetype
+   * for additional information on the restrictions.
+   *
+   * @default No tags are specified.
+   */
+  readonly tagSpecifications?: LaunchTemplateTagSpecification[];
```

Comment:
Tags will have to be implemented in core and in `cfn2ts`, to integrate with the aspect-based tagging (by doing it at the L1 level).

To start, let's take tagging out of this PR.

Reply:
I'd like to have tagging in, if possible. We need it for our use-case. 
I'll see how the ASG's tagging is done, and try to replicate it. I'm sure that I'll have questions...

Re: `cfn2ts`, is there a co-dependency here or is it okay to implement in CDK and then someone implements in `cfn2ts` later?

Reply:
Addition... Perhaps I'm being too clever/hacky, but I figured out a way to get the L2 LaunchTemplate to interact with tagging aspects without having to make any changes to CDK core.